Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -563,7 +563,8 @@ public class PBXProjGenerator {
     
     func getSources(sourceMetadata source: Source, path: Path, depth: Int = 0) throws -> (sourceFiles: [SourceFile], groups: [PBXGroup]) {
         // if we have a file, move it to children and use the parent as the path
-        let (children, path) = path.isFile ?
+        // pretend xcassets are files
+        let (children, path) = path.isFile || path.extension == "xcassets" ?
```

Comment:
Would it make sense to handle all directories with extensions like this?

Reply:
Good question -- I think some directories with extensions could actually be directories. Maybe we want a whitelist? It looks like the `xcproj` library handles a lot of these special ones, but `xcassets` are not handled by `xcproj` (which is why they have to be handled up here).

What do you think? What would we want in the whitelist?

Reply:
Yeah a whitelist would be good. I guess `.bundle` would go in there. Seems a cleaner solution to #128 as well

Reply:
I'll test that out

Reply:
Merging this in it's current state for now